[2024-12-23T09:30:29.953+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-12-23T09:30:29.992+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: streaming_pipeline.cdc_spark_streaming_to_datalake manual__2024-12-23T09:30:25.714754+00:00 [queued]>
[2024-12-23T09:30:30.004+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: streaming_pipeline.cdc_spark_streaming_to_datalake manual__2024-12-23T09:30:25.714754+00:00 [queued]>
[2024-12-23T09:30:30.005+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2024-12-23T09:30:30.021+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): cdc_spark_streaming_to_datalake> on 2024-12-23 09:30:25.714754+00:00
[2024-12-23T09:30:30.028+0000] {standard_task_runner.py:63} INFO - Started process 2931 to run task
[2024-12-23T09:30:30.030+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'streaming_pipeline', 'cdc_spark_streaming_to_datalake', 'manual__2024-12-23T09:30:25.714754+00:00', '--job-id', '81', '--raw', '--subdir', 'DAGS_FOLDER/pipline_stream.py', '--cfg-path', '/tmp/tmp_69dtq3v']
[2024-12-23T09:30:30.039+0000] {standard_task_runner.py:91} INFO - Job 81: Subtask cdc_spark_streaming_to_datalake
[2024-12-23T09:30:30.099+0000] {task_command.py:426} INFO - Running <TaskInstance: streaming_pipeline.cdc_spark_streaming_to_datalake manual__2024-12-23T09:30:25.714754+00:00 [running]> on host 0b4a5bf62610
[2024-12-23T09:30:30.200+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='streaming_pipeline' AIRFLOW_CTX_TASK_ID='cdc_spark_streaming_to_datalake' AIRFLOW_CTX_EXECUTION_DATE='2024-12-23T09:30:25.714754+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-12-23T09:30:25.714754+00:00'
[2024-12-23T09:30:30.201+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-12-23T09:30:30.221+0000] {logging_mixin.py:188} INFO - Checking JAR files...:  True
[2024-12-23T09:30:41.426+0000] {spark_streaming_to_dl.py:123} INFO - Spark session successfully created!
[2024-12-23T09:30:41.437+0000] {spark_streaming_to_dl.py:157} INFO - MinIO configuration is created successfully
[2024-12-23T09:30:43.367+0000] {spark_streaming_to_dl.py:176} INFO - Initial dataframe created successfully!
[2024-12-23T09:30:44.215+0000] {spark_streaming_to_dl.py:234} INFO - Final dataframe created successfully!
[2024-12-23T09:30:44.216+0000] {spark_streaming_to_dl.py:242} INFO - Streaming is being started...
[2024-12-23T09:31:40.861+0000] {local_task_job_runner.py:310} WARNING - State of this instance has been externally set to None. Terminating instance.
[2024-12-23T09:31:40.865+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2024-12-23T09:31:40.868+0000] {process_utils.py:132} INFO - Sending 15 to group 2931. PIDs of all processes in the group: [2945, 2931]
[2024-12-23T09:31:40.869+0000] {process_utils.py:87} INFO - Sending the signal 15 to group 2931
[2024-12-23T09:31:40.870+0000] {taskinstance.py:2611} ERROR - Received SIGTERM. Terminating subprocesses.
[2024-12-23T09:31:40.872+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-12-23T09:31:40.888+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/stream_processing/spark_streaming_to_dl.py", line 269, in run_all
    start_steaming(df_final, args.bucket_name, args.year, args.month)
  File "/opt/airflow/dags/stream_processing/spark_streaming_to_dl.py", line 253, in start_steaming
    return stream_query.awaitTermination()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/sql/streaming/query.py", line 221, in awaitTermination
    return self._jsq.awaitTermination()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/py4j/java_gateway.py", line 1321, in __call__
    answer = self.gateway_client.send_command(command)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/py4j/java_gateway.py", line 1038, in send_command
    response = connection.send_command(command)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/py4j/clientserver.py", line 511, in send_command
    answer = smart_decode(self.stream.readline()[:-1])
                          ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/socket.py", line 708, in readinto
    return self._sock.recv_into(b)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 2613, in signal_handler
    raise AirflowTaskTerminated("Task received SIGTERM signal")
airflow.exceptions.AirflowTaskTerminated: Task received SIGTERM signal
[2024-12-23T09:31:40.906+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=streaming_pipeline, task_id=cdc_spark_streaming_to_datalake, run_id=manual__2024-12-23T09:30:25.714754+00:00, execution_date=20241223T093025, start_date=20241223T093029, end_date=20241223T093140
[2024-12-23T09:31:40.935+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 81 for task cdc_spark_streaming_to_datalake ((psycopg2.errors.ForeignKeyViolation) insert or update on table "task_fail" violates foreign key constraint "task_fail_ti_fkey"
DETAIL:  Key (dag_id, task_id, run_id, map_index)=(streaming_pipeline, cdc_spark_streaming_to_datalake, manual__2024-12-23T09:30:25.714754+00:00, -1) is not present in table "task_instance".

[SQL: INSERT INTO task_fail (task_id, dag_id, run_id, map_index, start_date, end_date, duration) VALUES (%(task_id)s, %(dag_id)s, %(run_id)s, %(map_index)s, %(start_date)s, %(end_date)s, %(duration)s) RETURNING task_fail.id]
[parameters: {'task_id': 'cdc_spark_streaming_to_datalake', 'dag_id': 'streaming_pipeline', 'run_id': 'manual__2024-12-23T09:30:25.714754+00:00', 'map_index': -1, 'start_date': datetime.datetime(2024, 12, 23, 9, 30, 29, 992850, tzinfo=Timezone('UTC')), 'end_date': datetime.datetime(2024, 12, 23, 9, 31, 40, 904370, tzinfo=Timezone('UTC')), 'duration': 70}]
(Background on this error at: https://sqlalche.me/e/14/gkpj); 2931)
[2024-12-23T09:31:40.962+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=2931, status='terminated', exitcode=1, started='09:30:29') (2931) terminated with exit code 1
[2024-12-23T09:31:41.632+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=2945, status='terminated', started='09:30:29') (2945) terminated with exit code None
